# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.cpp
# Commit: 628ab886c5ec
# Full Hash: 628ab886c5ec1f4f4a926691d37b16674276af38
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-24 21:44:48
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Do not update variable that hold the media time if not playing in OnMuted. r=alwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D146196
# ==============================================================================

diff -r 65c876a5f665 -r 628ab886c5ec dom/media/mediasink/AudioSinkWrapper.cpp
--- a/dom/media/mediasink/AudioSinkWrapper.cpp	Tue May 24 13:09:10 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.cpp	Tue May 24 13:09:10 2022 +0000
@@ -169,8 +169,10 @@
     if (mAudioSink) {
       LOG("AudioSinkWrapper muted, shutting down AudioStream.");
       mAudioSinkEndedPromise.DisconnectIfExists();
-      mPlayDuration = mAudioSink->GetPosition();
-      mPlayStartTime = TimeStamp::Now();
+      if (IsPlaying()) {
+        mPlayDuration = mAudioSink->GetPosition();
+        mPlayStartTime = TimeStamp::Now();
+      }
       Maybe<MozPromiseHolder<MediaSink::EndedPromise>> rv =
           mAudioSink->Shutdown(ShutdownCause::Muting);
       MOZ_ASSERT(rv.isSome());
